### PR TITLE
[MAINTENANCE] Turn off metric calculation progress bars in `RuleBasedProfiler` and `DataAssistant` workflows

### PR DESCRIPTION
**Pull request by cdkini**
Changes proposed in this pull request:
- Removal of progress bars for metric calculations to reduce clutter in Jupyter
- Patch deprecation warning within RBP flow to improve readability
- https://www.loom.com/share/4d3c938bdc994aff8f756dab15f2d350

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/util.py [hunk 1]
```diff
@@ -1489,3 +1490,28 @@ def pandas_series_between_inclusive(
...
+
+
+def numpy_quantile(
+    a: np.ndarray, q: float, method: str, axis: Optional[int] = None
+) -> np.ndarray:
+    """
+    As of NumPy 1.21.0, the 'interpolation' arg in quantile() has been renamed to `mode`.
+    Source: https://numpy.org/doc/stable/reference/generated/numpy.quantile.html
+    """
+    quantile: np.ndarray
```
Comment (Shinnnyshinshin) [inline great_expectations/util.py]: ❤️

File: great_expectations/validator/validator.py [hunk 2]
```diff
@@ -228,6 +230,27 @@ def __dir__(self):
...
+        return enable
+
+    @property
+    def show_progress_bars(self) -> bool:
+        return self._show_progress_bars
+
+    @show_progress_bars.setter
+    def show_progress_bars(self, enable: bool) -> None:
+        self._show_progress_bars = enable
+
```
Comment (Shinnnyshinshin) [inline great_expectations/validator/validator.py:253]: omg this is beautiful. thank you for this @cdkini

**Review by Shinnnyshinshin — APPROVED** [review]
Looks great

File: great_expectations/rule_based_profiler/helpers/util.py [hunk 3]
```diff
@@ -469,17 +472,17 @@ def compute_quantiles(
...
     quantile_statistic_interpolation_method: str,
 ) -> NumericRangeEstimationResult:
-    lower_quantile = np.quantile(
-        metric_values,
+    lower_quantile = numpy_quantile(
+        a=metric_values,
         q=(false_positive_rate / 2),
         axis=0,
-        interpolation=quantile_statistic_interpolation_method,
+        method=quantile_statistic_interpolation_method,
```
Comment (alexsherstinsky) [inline great_expectations/rule_based_profiler/helpers/util.py:479]: @cdkini Is this not going to fail tests for `Python < 3.9`?..  Thanks.

File: great_expectations/util.py [hunk 4]
```diff
@@ -1489,3 +1490,28 @@ def pandas_series_between_inclusive(
         metric_series = series.between(min_value, max_value, inclusive=True)
 
     return metric_series
+
+
+def numpy_quantile(
+    a: np.ndarray, q: float, method: str, axis: Optional[int] = None
+) -> np.ndarray:
+    """
+    As of NumPy 1.21.0, the 'interpolation' arg in quantile() has been renamed to `mode`.
```
Comment (alexsherstinsky) [inline great_expectations/util.py]: @cdkini Not `mode`, but `method`, correct?  Thanks.

**Review by alexsherstinsky — CHANGES_REQUESTED** [review]
A few minor comments...

File: great_expectations/util.py (hunk 4 above)
Comment (cdkini) [reply great_expectations/util.py]: you are right!

File: great_expectations/rule_based_profiler/helpers/util.py (hunk 3 above)
Comment (cdkini) [reply great_expectations/rule_based_profiler/helpers/util.py:479]: I don't believe so? Could you elaborate why you think so?

**Review by alexsherstinsky — APPROVED** [review]
LGTM

**Pull request merged**